src/functions/transactor.ts

```diff
@@ -141,13 +167,29 @@ export const transactor = (
         return result;
       } catch (e: any) {
         if (DEBUG) console.log(e);
+
+        const err = e as TRawTxError;
+
+        const extractedReason = err.data?.message?.match(/reverted with reason string \'(.*?)\'/);
         // Accounts for Metamask and default signer on all networks
-        notification.error({
+
+        let notificationMessage: ArgsProps = {
           message: 'Transaction Error',
-          description: e?.message,
-        });
+          description: err.message,
+        };
+
+        if (extractedReason && extractedReason.length > 0) {
+          notificationMessage.description = extractedReason[1];
+        }
+
+        if (filterErrorMessage) {
+          notificationMessage = filterErrorMessage(err, notificationMessage);
+        }
+
+        notification.error(notificationMessage);
 
-        if (throwOnError) throw e;
+        if (throwOnError)
+          throw new TransactorError({ message: notificationMessage.description?.toString(), rawError: err });

```

Comment:
`notificationMessage.description` it's `ReactNode` type and optional coming from the `ArgsProps` from `Antd Lib`.